python/rikai/torch/data.py

```diff
@@ -70,17 +75,19 @@ class Dataset(IterableDataset):
 
     def __init__(
         self,
-        uri: Union[str, Path],
+        uri_or_df: Union[str, Path, 'pyspark.sql.DataFrame'],
         columns: List[str] = None,
         transform: Callable = RikaiToTensor(),
+        parquet_row_group_size_bytes = None,
     ):
         super().__init__()
-        self.uri = str(uri)
+        self.uri_or_df = uri_or_df
         self.columns = columns
         self._transform = transform
+        self.parquet_row_group_size_bytes = parquet_row_group_size_bytes
 
     def __repr__(self) -> str:
-        return f"Dataset(torch, {self.uri}, columns={self.columns})"
+        return f"Dataset(torch, {self.uri_or_df}, columns={self.columns})"
```

Comment:
nit, is df printable with meaningful info here?

Reply:
this pyspark DataFrame's repr:

```
    def __repr__(self):
        if not self._support_repr_html and self.sql_ctx._conf.isReplEagerEvalEnabled():
            vertical = False
            return self._jdf.showString(
                self.sql_ctx._conf.replEagerEvalMaxNumRows(),
                self.sql_ctx._conf.replEagerEvalTruncate(), vertical)
        else:
            return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
```